internal/bminventory/inventory.go

```diff
@@ -2987,6 +2987,31 @@ func (b *bareMetalInventory) processDiskSpeedCheckResponse(ctx context.Context,
 	return b.hostApi.SetDiskSpeed(ctx, h, diskPerfCheckResponse.Path, diskPerfCheckResponse.IoSyncDuration, exitCode, nil)
 }
 
+func (b *bareMetalInventory) updateDomainNameResolutionResponse(ctx context.Context, host *models.Host, domainResolutionResponseReporter string) error {
+	var (
+		err                      error
+		domainResolutionResponse models.DomainResolutionResponse
+	)
+	log := logutil.FromContext(ctx, b.log)
+	log.Debugf("The address for domain name resolotion on host %s are: %s", host.ID.String(), domainResolutionResponse)
+
+	if err = json.Unmarshal([]byte(domainResolutionResponseReporter), &domainResolutionResponse); err != nil {
+		log.WithError(err).Warnf("Json unmarshal domain name resolution of host %s", host.ID.String())
+		return err
+	}
+	if len(domainResolutionResponse.Resolutions) == 0 {
```

Comment:
This logic should not be done here, but as part of a validation.

Reply:
Other steps also perform validations on the agent input before storing them in the database

Reply:
e.g. https://github.com/openshift/assisted-service/blob/1d9e078f900e159d0b7afc09b9eb78f6d58648d6/internal/bminventory/inventory.go#L2874-L2877

Reply:
@ori-amizur seems to me that if the input we are getting, from the host is invalid, why should we put it in the DB and then need to fail this validation every time?

Reply:
We should just try to unmarshal, but if the response is syntactically valid, we should not validate the response inside.  This logic already exists in the validation. Why twice?

Reply:
Agree with Ori, moved all response logic validation to the validator function insted.